include/roboteam_robothub/SerialDeviceManager.h

```diff
@@ -16,11 +16,11 @@ class SerialDeviceManager {
     explicit SerialDeviceManager() = default;
     explicit SerialDeviceManager(const std::string &deviceName);
     bool ensureDeviceOpen();
-    bool writeToDevice(packed_protocol_message packet);
+    bool writeToDevice(const RobotCommandPayload& command);
     void openDevice();
-    std::shared_ptr<packed_robot_feedback> readDevice();
-    std::shared_ptr<packed_robot_feedback> mostRecentFeedback = nullptr;
-    std::shared_ptr<packed_robot_feedback> getMostRecentFeedback() const;
+    std::optional<RobotFeedbackPayload> readDevice();
+    std::optional<RobotFeedbackPayload> mostRecentFeedback = std::nullopt;
+    [[nodiscard]] std::optional<RobotFeedbackPayload> getMostRecentFeedback() const;
```

Comment:
What does this function actually do? I thought that feedback packets were forwarded immediately to AI

Reply:
If you look at the code, I'm just adhering to legacy here. It's called almost immediately after readDevice()

Reply:
So they are forwarded immediately to the AI; it's just to have a failsafe.